File: languageComparison/getSubseries/testing/ruby/getSubseries.rb
# getSubseries.rb

def getSubseries(series, threshold)
	working = []
	series.inject [] do |reduction, current|
		working.push current
		working.shift while working.inject(0, &:+) > threshold

		workingIsBetterForLength	= working.length > reduction.length
		workingIsBetterForTotal		= working.length == reduction.length && working.inject(0, &:+) > reduction.inject(0, &:+)
		
		workingIsBetterForLength || workingIsBetterForTotal ? working.clone : reduction
	end
end

File: languageComparison/getSubseries/testing/python/getSubseries.py
# getSubseries.py

def getSubseries(series,threshold):
	import functools
	working = []
	def reduceCallback(reduction, current):
		working.append(current)
		while sum(working) > threshold:
			working.pop(0)

		workingIsBetterForLength= len(working) > len(reduction)
		workingIsBetterForTotal	= len(working) == len(reduction) and sum(working) > sum(reduction)

		return list(working) if workingIsBetterForLength or workingIsBetterForTotal else reduction

	return functools.reduce(reduceCallback, series, [])
